content/events-api/security.md

```diff
@@ -6,9 +6,50 @@ The signature is forwarded through the request header `X-Akeneo-Request-Signatur
 
 You can verify the signature by following these steps:
 
-1. Get the original signature and timestamp, extract them from the following request headers: `X-Akeneo-Request-Signature` and `X-Akeneo-Request-Timestamp`
-2. Prepare the event payload which will be signed by concatenating the timestamp and the request body with a dot `.`
-3. Generate a hash signature with the HMAC SHA256 algorithm using the secret as the secret key.
-4. Compare the original and generated signature, you are encouraged to use a string comparison function that is safe to timing attack, like [hash_equals](https://www.php.net/manual/en/function.hash-equals.php) in PHP.
+1. Get the original signature and timestamp from the request headers:
+   `X-Akeneo-Request-Signature` and `X-Akeneo-Request-Timestamp`.
+2. Prepare the event payload by concatenating the timestamp and the request body with a dot.
+3. Generate a hash signature (HMAC SHA256) of the event payload with the help of the _secret_.
+4. Compare the original and generated signature.
 
-To see how to implement these steps in a Symfony application you can follow the [Quick start my first webhook](/getting-started/quick-start-my-first-webhook/welcome.html) guide.
+Example of signature validation with PHP
+
+```php
+
+$secret = "3ha6eonoa9icsckw8kccos084w0c0000g08g40oo4kww0gc8w4";
+
+// X-Akeneo-Request-Signature
+$originalSignature =
+  "d4582dd13a7b654c212d70b4c705e0ab0c0e6eb7f6524080f4d39121f88e6061";
+// X-Akeneo-Request-Timestamp
+$timestamp = "1602565368";
+
+$requestBody = '{"action":"product.updated", ...}';
+
+// Prepare the event payload.
+$signedPayload = $timestamp . "." . $requestBody;
+
+// Generate a hash signature.
+$generatedSignature = hash_hmac("sha256", $signedPayload, $secret);
+
+// Compare the original and generated signature.
+if (false === hash_equals($originalSignature, $generatedSignature)) {
+  throw new \Exception("Invalid signature");
+}
+```
+
+::: info
+You are encouraged to use a string comparison function that is safe to timing attack, like [hash_equals](https://www.php.net/manual/en/function.hash-equals.php) in PHP.
+:::
+
+You can also use the `X-Akeneo-Request-Timestamp` to validate that the event request is not too old.
+
+```php
+
+// X-Akeneo-Request-Timestamp
+$timestamp = "1602565368";
+
+if (time() - (int) $timestamp > 300) {
+  throw new \Exception("Request is too old (> 5min)");
+}
+```
```

Comment:
Are you sure we should recommend it ?

Reply:
This is a more a general security recommendation, you can see this in the documentation of other webhook providers.
Because if you request take more than 5 minutes to reach the server there is something seriously wrong on the network.

Reply:
I get it, but I always want to be careful about writing too much documentation on things that are not useful to most of our users. Is this really something related to security ?
A one-liner on what could be done with this header would be better than showing them how to handle it, IMHO.

Reply:
I can remove the code snippet if that's the issue. There is nothing more to say that the sentence already here.
If you want to see the full explanation from the Stripe documentation on "Replay attacks" https://stripe.com/docs/webhooks/signatures#replay-attacks

Reply:
I don't think we are concerned by replay attacks, we are not touching on banking stuff at all.
If you think that the snippet is relevant anyway, you can keep it.